### PR TITLE
[Blog] Adding MacOS Arm64 Core Snapshot Artifact Blog Post

### PR DESCRIPTION
**Pull request by peterzhuamazon**
### Description
Adding MacOS Arm64 Core Snapshot Artifact Blog Post
 
### Issues Resolved
https://github.com/opensearch-project/project-website/issues/3131

### Check List
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the BSD-3-Clause License.

### CODE REVIEW

**Comment by peterzhuamazon** [comment]
Adding @pajuric to take a look on the blog content, release timing, etc.

Thanks!

**Comment by peterzhuamazon** [comment]
Hi @pajuric @krisfreedain @natebower any update on this?

Thanks.

**Comment by pajuric** [comment]
@peterzhuamazon Hi Peter - I have the blog in review with my team, and then it will go to Nathan Bower. Right now, I have two blogs that ahead of your blog, so it will likely be a week before it goes out.

**Comment by peterzhuamazon** [comment]
> @peterzhuamazon Hi Peter - I have the blog in review with my team, and then it will go to Nathan Bower. Right now, I have two blogs that ahead of your blog, so it will likely be a week before it goes out.

Thanks @pajuric !

I will change it to Oct 21 for now.

**Comment by natebower** [comment]
@kolchfa-aws will perform doc review prior to editorial.

**Comment by kolchfa-aws** [comment]
@peterzhuamazon Could you verify that my changes preserved the technical accuracy? Once you do, we'll move the blog to editorial review. Thanks!

File: _posts/2024-09-30-Boosting-OpenSearch-Development-on-MacOS-with-Apple-Silicon.md [hunk 1]
```diff
@@ -1,46 +1,44 @@
 ---
 layout: post
-title:  Boosting OpenSearch Development on MacOS with Apple Silicon
+title:  Streamline your OpenSearch development workflow on macOS with Apple silicon
 authors:
 - zhujiaxiang
-- ssingh
+- sisurab
+- kolchfa
 date: 2024-09-30
```
Comment (peterzhuamazon) [inline _posts/2024-09-30-Boosting-OpenSearch-Development-on-MacOS-with-Apple-Silicon.md:8]: Would need to change this date and the file name to corresponding blog release date once settled.

**Comment by peterzhuamazon** [comment]
> @peterzhuamazon Could you verify that my changes preserved the technical accuracy? Once you do, we'll move the blog to editorial review. Thanks!

Hi @kolchfa-aws ,

Thanks for the changes and I have reviewed those.
No issues from my side regarding technical accuracy and improves the blog a lot. Thanks!

File: _posts/2024-09-30-Boosting-OpenSearch-Development-on-MacOS-with-Apple-Silicon.md [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+meta_keywords: opensearch open source, MacOS, ARM64, Apple Silicon, developer friendly, architecture support, platform support
+meta_description: Learn how to simplify your OpenSearch development workflow on macOS with Apple silicon
+---
+
+OpenSearch is an Apache-2.0 licensed open-source project supported by an engaging community of users across the globe. Building on its commitment to openness and growth, OpenSearch recently became part of the newly formed [OpenSearch Software Foundation](https://foundation.opensearch.org/), a community-driven initiative under the [Linux Foundation](https://www.linuxfoundation.org/). This shift has allowed us to enhance the developer experience by expanding platform and architecture support, especially for developers using macOS ARM64 machines. Previously, setting up OpenSearch cluster and running integration tests required a Linux environment or Docker containers, adding complexity and overhead to the process. Now, you can run and test OpenSearch plugins natively on macOS, simplifying workflows and reducing development time.
```
Comment (natebower) [inline _posts/2024-09-30-Boosting-OpenSearch-Development-on-MacOS-with-Apple-Silicon.md:17]:
```suggestion
OpenSearch is an Apache 2.0--licensed open-source project supported by an engaged community of users across the globe. Building on its commitment to openness and growth, OpenSearch recently became part of the newly formed [OpenSearch Software Foundation](https://foundation.opensearch.org/), a community-driven initiative under the [Linux Foundation](https://www.linuxfoundation.org/). This shift has allowed us to enhance the developer experience by expanding platform and architecture support, especially for developers using macOS ARM64 machines. Previously, setting up an OpenSearch cluster and running integration tests required a Linux environment or Docker containers, adding complexity and overhead to the process. Now you can run and test OpenSearch plugins natively on macOS, simplifying workflows and reducing development time.
```

File: _posts/2024-09-30-Boosting-OpenSearch-Development-on-MacOS-with-Apple-Silicon.md [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+
+We're excited to announce that the OpenSearch core snapshot tarball artifact, built specifically for macOS ARM64 machines, is now publicly available. If you're developing on these machines, especially the latest Mac computers with Apple silicon, we encourage you to give it a try.
+
+In this blog post, we'll discuss accessing the new snapshot artifact and using it to streamline your OpenSearch and OpenSearch Dashboards development workflow.
+
+
+## Prerequisite
+
+You can use the artifact described in this post on a [Mac computer with Apple silicon](https://support.apple.com/en-us/116943) (ARM64 architecture). For older models with the Intel processor, you can use the [OpenSearch core snapshot artifact for macOS X64 Host](https://github.com/opensearch-project/opensearch-build/issues/2216).
```
Comment (natebower) [inline _posts/2024-09-30-Boosting-OpenSearch-Development-on-MacOS-with-Apple-Silicon.md:26]:
```suggestion
You can use the artifact described in this post on a [Mac computer with Apple silicon](https://support.apple.com/en-us/116943) (ARM64 architecture). For older models with an Intel processor, you can use the [OpenSearch core snapshot artifact for macOS X64 Host](https://github.com/opensearch-project/opensearch-build/issues/2216).
```

File: _posts/2024-09-30-Boosting-OpenSearch-Development-on-MacOS-with-Apple-Silicon.md [hunk 4]
```diff
@@ -0,0 +1,234 @@
...
+
+# Extract the tarball
+tar -xzvf opensearch-min-3.0.0-SNAPSHOT-darwin-arm64-latest.tar.gz
+
+# Start the cluster
+cd opensearch-3.0.0-SNAPSHOT
+./bin/opensearch
+```
+
+Once the cluster is started, you can make a test API call to verify that the cluster is working:
```
Comment (natebower) [inline _posts/2024-09-30-Boosting-OpenSearch-Development-on-MacOS-with-Apple-Silicon.md:55]: "make" => "send"?

File: _posts/2024-09-30-Boosting-OpenSearch-Development-on-MacOS-with-Apple-Silicon.md [hunk 5]
```diff
@@ -0,0 +1,234 @@
...
+
+```bash
+curl http://localhost:9200
+```
+
+The response should look similar to the following:
```
Comment (natebower) [inline _posts/2024-09-30-Boosting-OpenSearch-Development-on-MacOS-with-Apple-Silicon.md:61]:
```suggestion
The response should appear similar to the following:
```

File: _posts/2024-09-30-Boosting-OpenSearch-Development-on-MacOS-with-Apple-Silicon.md [hunk 6]
```diff
@@ -0,0 +1,234 @@
...
+  },
+  "tagline" : "The OpenSearch Project: https://opensearch.org/"
+}
+```
+
+## Run OpenSearch plugin integration tests 
+
+Follow this procedure to run integration tests for an OpenSearch plugin. In this example, you'll run tests for the [OpenSearch Alerting plugin](https://github.com/opensearch-project/alerting) on an M1 MacBook.
+
+Clone the alerting repository and change directory to the repository root:
```
Comment (natebower) [inline _posts/2024-09-30-Boosting-OpenSearch-Development-on-MacOS-with-Apple-Silicon.md:87]:
```suggestion
Clone the alerting repository and change the directory to the repository root:
```

File: _posts/2024-09-30-Boosting-OpenSearch-Development-on-MacOS-with-Apple-Silicon.md [hunk 7]
```diff
@@ -0,0 +1,234 @@
...
+
+You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.
+
+For more on this, please refer to https://docs.gradle.org/8.5/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
+
+BUILD SUCCESSFUL in 22m 48s
+28 actionable tasks: 25 executed, 3 up-to-date
+```
+
+**Note**: Running integration tests with the new artifact also works for OpenSearch version 1.3.19 (when you switch to the 1.3 branch) or any version 3.0.0 and later (when you switch to main branch).
```
Comment (natebower) [inline _posts/2024-09-30-Boosting-OpenSearch-Development-on-MacOS-with-Apple-Silicon.md:179]:
```suggestion
**Note**: Running integration tests with the new artifact also works for OpenSearch version 1.3.19 (when you switch to the 1.3 branch) or version 3.0.0 and later (when you switch to the main branch).
```

File: _posts/2024-09-30-Boosting-OpenSearch-Development-on-MacOS-with-Apple-Silicon.md [hunk 8]
```diff
@@ -0,0 +1,234 @@
...
+   │ info [o.o.h.AbstractHttpServerTransport] [macos-arm64-host] publish_address {127.0.0.1:9200}, bound_addresses {[::1]:9200}, {127.0.0.1:9200}
+   │ info [o.o.n.Node] [macos-arm64-host] started
+......
+```
+
+Now that OpenSearch is running in the background, you can start developing OpenSearch Dashboards on your macOS ARM64 host.
+
+## Next steps
+
+If you're a developer working on a macOS ARM64 machine, we encourage you to test the artifact and share your feedback. If you have any questions, feel free to join us on our [public slack channel](https://opensearch.org/slack.html) or in the [GitHub repository](https://github.com/opensearch-project).
```
Comment (natebower) [inline _posts/2024-09-30-Boosting-OpenSearch-Development-on-MacOS-with-Apple-Silicon.md:233]:
```suggestion
If you're a developer working on a macOS ARM64 machine, we encourage you to test the artifact and share your feedback. If you have any questions, feel free to join us on our [public Slack channel](https://opensearch.org/slack.html) or in the [GitHub repository](https://github.com/opensearch-project).
```

**Review by natebower — COMMENTED** [review]
@peterzhuamazon @kolchfa-aws Editorial review complete. Looking good! Please see my comment and changes and let me know if you have any questions. Thanks!

Cc: @pajuric

File: _posts/2024-09-30-Boosting-OpenSearch-Development-on-MacOS-with-Apple-Silicon.md (hunk 4 above)
Comment (kolchfa-aws) [reply _posts/2024-09-30-Boosting-OpenSearch-Development-on-MacOS-with-Apple-Silicon.md:55]: You make a call but you send a request 😄 so leaving as is.

**Comment by kolchfa-aws** [comment]
@pajuric Editorial comments implemented. Could you review/update the meta, and we'll be ready to publish.

File: _posts/2024-09-30-Boosting-OpenSearch-Development-on-MacOS-with-Apple-Silicon.md [hunk 9]
```diff
@@ -0,0 +1,234 @@
...
+
+```bash
+git clone https://github.com/opensearch-project/alerting.git
+cd alerting
+```
+
+First, try running integration tests on an older version of OpenSearch that doesn't contain the core snapshot for macOS ARM64 (in this example, 2.14.0):
+
+```bash
+# Checkout 2.14
```
Comment (kolchfa-aws) [inline _posts/2024-09-30-Boosting-OpenSearch-Development-on-MacOS-with-Apple-Silicon.md:97]:
```suggestion
# Check out 2.14
```

File: _posts/2024-09-30-Boosting-OpenSearch-Development-on-MacOS-with-Apple-Silicon.md [hunk 10]
```diff
@@ -0,0 +1,234 @@
...
+For more on this, please refer to https://docs.gradle.org/8.5/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
+
+BUILD FAILED in 1m 31s
+14 actionable tasks: 13 executed, 1 up-to-date
+```
+
+Then switch to the 2.16.0 version, in which the macOS ARM64 snapshot is available, and try again:
+
+```bash
+# Checkout 2.16
```
Comment (kolchfa-aws) [inline _posts/2024-09-30-Boosting-OpenSearch-Development-on-MacOS-with-Apple-Silicon.md:140]:
```suggestion
# Check out 2.16
```

File: _posts/2024-09-30-Boosting-OpenSearch-Development-on-MacOS-with-Apple-Silicon.md [hunk 11]
```diff
@@ -0,0 +1,234 @@
...
+  },
+  "tagline" : "The OpenSearch Project: https://opensearch.org/"
+}
+```
+
+## Run OpenSearch plugin integration tests 
+
+Follow this procedure to run integration tests for an OpenSearch plugin. In this example, you'll run tests for the [OpenSearch Alerting plugin](https://github.com/opensearch-project/alerting) on an M1 MacBook.
+
+Clone the alerting repository and change the directory to the repository root:
```
Comment (kolchfa-aws) [inline _posts/2024-09-30-Boosting-OpenSearch-Development-on-MacOS-with-Apple-Silicon.md:87]:
```suggestion
Clone the alerting repository and change directory to the repository root:
```

File: _posts/2024-10-21-Boosting-OpenSearch-Development-on-MacOS-with-Apple-Silicon.md [hunk 12]
```diff
@@ -0,0 +1,234 @@
...
+- zhujiaxiang
+- sisurab
+- kolchfa
+date: 2024-10-21
+categories:
+  - releases
+  - technical-posts
+excerpt: We're excited to announce that the OpenSearch core snapshot tarball artifact, built specifically for macOS ARM64 machines, is now publicly available. If you're developing on the latest Mac computers with Apple silicon, we encourage you to give it a try and experience the streamlined workflow.
+meta_keywords: opensearch open source, MacOS, ARM64, Apple Silicon, developer friendly, architecture support, platform support
+meta_description: Learn how to simplify your OpenSearch development workflow on macOS with Apple silicon
```
Comment (pajuric) [inline _posts/2024-10-21-Boosting-OpenSearch-Development-on-MacOS-with-Apple-Silicon.md:14]: @peterzhuamazon  - Updated meta: 
_**meta_keywords:**_  OpenSearch core snapshot,  macOS ARM64, OpenSearch Dashboards, development workflow on macOS with Apple silicon

_**meta_description:**_  Learn how to access the new OpenSearch core snapshot artifact for macOS ARM64 and use it to simplify your development process and streamline your workflow on Mac computers with Apple silicon.

**Comment by peterzhuamazon** [comment]
Hi @pajuric , 

I have updated the meta description and keywords now.

Thanks.

File: _posts/2024-10-21-Boosting-OpenSearch-Development-on-MacOS-with-Apple-Silicon.md (hunk 12 above)
Comment (peterzhuamazon) [reply _posts/2024-10-21-Boosting-OpenSearch-Development-on-MacOS-with-Apple-Silicon.md]: Updated now.

**Comment by pajuric** [comment]
@peterzhuamazon - I am going to see if my team can push it live now.

**Comment by pajuric** [comment]
@nateynateynate - Can you help push this blog live and update the date to today?

File: _posts/2024-10-21-Boosting-OpenSearch-Development-on-MacOS-with-Apple-Silicon.md [hunk 13]
```diff
@@ -0,0 +1,234 @@
+---
+layout: post
+title:  Streamline your OpenSearch development workflow on macOS with Apple silicon
+authors:
+- zhujiaxiang
+- sisurab
+- kolchfa
+date: 2024-10-21
```
Comment (kolchfa-aws) [inline _posts/2024-10-21-Boosting-OpenSearch-Development-on-MacOS-with-Apple-Silicon.md:8]:
```suggestion
date: 2024-10-22
```

**Review by kolchfa-aws — APPROVED** [review]
LGTM

**Pull request merged**